From 549a119eb2513570952238953451b072d88bb80f Mon Sep 17 00:00:00 2001
From: Andrew Wildman <apw4@uw.edu>
Date: Mon, 11 Oct 2021 10:01:18 -0700
Subject: [PATCH] Fix parallel compilation on Umpire

---
 external/umpire.cmake | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/external/umpire.cmake b/external/umpire.cmake
index bed350f0..6d278c02 100644
--- a/external/umpire.cmake
+++ b/external/umpire.cmake
@@ -135,8 +135,6 @@ else()
             BUILD_BYPRODUCTS ${UMPIRE_BUILD_BYPRODUCTS}
             #--Install step---------------
             INSTALL_COMMAND ${CMAKE_COMMAND} -E echo "Umpire will be installed during TiledArray's installation."
-            #--Custom targets-------------
-            STEP_TARGETS build
             )
 
     # do install of Umpire as part of building TiledArray's install target
@@ -151,7 +149,7 @@ else()
             ")
 
     # Add Umpire dependency to External
-    add_dependencies(External-tiledarray Umpire-build)
+    add_dependencies(External-tiledarray Umpire)
 
     set(_UMPIRE_INSTALL_DIR ${EXTERNAL_INSTALL_DIR})
 
-- 
2.19.2

