--- include/lapack/config.h
+++ include/lapack/config.h
@@ -64,7 +64,7 @@
      * This isn't compatible as a return type from extern C functions,
      * so it may generate compiler warnings or errors. */
     #include <complex>
-    typedef std::complex<float>  lapack_complex_float;
+    //typedef std::complex<float>  lapack_complex_float;
     typedef std::complex<double> lapack_complex_double;
     #define lapack_complex_float_real(z)  ((z).real())
     #define lapack_complex_float_imag(z)  ((z).imag())
@@ -85,8 +85,8 @@
 #endif
 
 /* define so we can check later with ifdef */
-#define lapack_complex_float  lapack_complex_float
-#define lapack_complex_double lapack_complex_double
+//#define lapack_complex_float  lapack_complex_float
+//#define lapack_complex_double lapack_complex_double
 
 #ifdef __cplusplus
 extern "C" {
