--- f77blas.h	2017-06-27 15:30:32.199229302 -0700
+++ f77blas.h.new	2017-06-27 15:13:52.094038837 -0700
@@ -647,6 +647,7 @@
 
 /* Lapack routines */
 
+/*
 int BLASFUNC(sgetf2)(blasint *, blasint *, float  *, blasint *, blasint *, blasint *);
 int BLASFUNC(dgetf2)(blasint *, blasint *, double *, blasint *, blasint *, blasint *);
 int BLASFUNC(qgetf2)(blasint *, blasint *, xdouble *, blasint *, blasint *, blasint *);
@@ -739,6 +740,7 @@
 FLOATRET  BLASFUNC(slamc3)(float *, float *);
 double    BLASFUNC(dlamc3)(double *, double *);
 xdouble   BLASFUNC(qlamc3)(xdouble *, xdouble *);
+*/
 
 /* BLAS extensions */
 
