--- lapack.h    2020-05-19 15:07:57.448525158 -0700
+++ lapack.h.new    2020-05-19 15:29:25.794509843 -0700
@@ -10646,7 +10646,7 @@
     lapack_int* iwork, lapack_int* IFAIL,
     lapack_int* info );
 
-#define LAPACK_csyr LAPACK_GLOBAL(csyr,CSYR)
+// #define LAPACK_csyr LAPACK_GLOBAL(csyr,CSYR)
 void LAPACK_csyr(
     char const* uplo,
     lapack_int const* n,
@@ -10654,7 +10654,7 @@
     lapack_complex_float const* X, lapack_int const* incx,
     lapack_complex_float* A, lapack_int const* lda );
 
-#define LAPACK_zsyr LAPACK_GLOBAL(zsyr,ZSYR)
+// #define LAPACK_zsyr LAPACK_GLOBAL(zsyr,ZSYR)
 void LAPACK_zsyr(
     char const* uplo,
     lapack_int const* n,
